styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java

```diff
@@ -202,9 +204,10 @@ private void logAuthorization(WorkflowId workflowId, String serviceAccount, bool
           accessMessage, serviceAccount, workflowId.toKey(), enforce, cached ? CACHE_HIT : CACHE_MISS);
     }
 
-    private String serviceAccountProjectId(String serviceAccount) {
+    private String serviceAccountProjectId(WorkflowId workflowId, String serviceAccount) {
       final Matcher matcher = USER_CREATED_SERVICE_ACCOUNT_PATTERN.matcher(serviceAccount);
       if (!matcher.matches()) {
+        log.info("Non user created service account {} is used in workflow {}", serviceAccount, workflowId.toKey());
```

Comment:
This log message is a bit misleading. It sounds like a non-user created SA is used, but we reject the request, so it will not be used.